pymine/logic/parsers/minecraft/blockpredicate.py

```diff
@@ -0,0 +1,15 @@
+from pymine.data.block_states import BLOCK_STATES
+from pymine.api.errors import ParsingError
+from pymine.api.abc import AbstractParser
+
+class BlockPredicate(AbstractParser):
+    def __init__(self) -> None:
+        pass
+
+    def parse(self, s: str, p:[Player, None]) -> tuple:
+        section = s.split()[0]
+        try:
+            return len(section), BLOCK_STATES[section]
+        except BaseException:
+            pass
```

Comment:
Should be a newline following de-dentation if the dedentation isn't part of something like `try: except:` or a set of `if`/`elif`/`else` statements